pkg/apis/v1beta1/nodepool.go

```diff
@@ -190,6 +190,12 @@ type NodePool struct {
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
+// We need to bump the NodePoolHashVersion under these condition when we make an update the NodePool CRD
+// 1. A NodePool adds default values to existing fields that are included in the hash calculations
+// 2. A NodePool field is added to the hash calculation with an already-set value
+// 3. A NodePool field is removed from the hash calculations
```

Comment:
nit: Since it's called NodePool..... you can probably just remove the NodePool bit
```suggestion
// 1. A change to the default value to existing fields that are already hashed 
// 2. A field is added to the hash calculation with an already-set value
// 3. A field is removed from the hash calculations
```